pkg/controller/networkaddonsconfig/networkaddonsconfig_controller.go

```diff
@@ -498,6 +495,97 @@ func (r *ReconcileNetworkAddonsConfig) stopTrackingObjects() {
 	r.statusManager.SetFromPods()
 }
 
+func updateObjectstLabels(crLables map[string]string, objs []*unstructured.Unstructured) ([]*unstructured.Unstructured, error) {
+	var err error
+	for _, obj := range objs {
+		labels := obj.GetLabels()
+		if labels == nil {
+			labels = map[string]string{}
+		}
+		// Label objects with version of the operator they were created by
+		labels[cnaov1.SchemeGroupVersion.Group+"/version"] = operatorVersionLabel
+
+		labels, err = updateLabels(obj, labels, crLables)
+		if err != nil {
+			return objs, err
+		}
+
+		labels, err = updateManagedByLabel(obj, labels, crLables)
+		if err != nil {
+			return objs, err
+		}
+
+		obj.SetLabels(labels)
+	}
+
+	return objs, nil
+}
+
+func updateLabels(obj *unstructured.Unstructured, labels, crLables map[string]string) (map[string]string, error) {
+	var err error
+	const partOfLabelKey = "app.kubernetes.io/part-of"
```

Comment:
perhaps this should be declared in a more general place? it's would make it easier to be used by unit tests later on.

Reply:
btw unit tests does not check reconcile
the one that do check is e2e  test/check/check.go
```

checkForGenericDeployment
checkForComponent
checkConfigChange
```
Those check for example the version label that was already exists, in the same code that the PR updated

Reply:
is there a problem to create a `networkaddonsconfig_controller_test.go` and add the unit tests for updateObjectsLabels(), updateObjLabels(), updateLabelByKey(), updateManagedByLabel()?

Reply:
I would suggest to start with the e2e tests, the infrastructure for them is ready
on following PRs we can add the unit tests, as it seems there isn't even infra for checking the controller in unit test
is it ok ?